w3-filecoin.md

```diff
@@ -304,265 +355,267 @@ If the added piece is invalid, details on failing reason is also reported:
+An [Aggregator] MUST issue a signed receipt to acknowledge the received request. The receipt MUST contain a `fx.join` [effect] linking to [`piece/accept`] task that MUST either succeed with [`InclusionProof`] or fail with an error describing the reason.
 
 ```json
 {
-  "ran": "bafy...invocation",
+  "ran": "bafy...pieceOffer",
   "out": {
-    "ok": {}
+    "ok": {
+      /* commitment proof for piece */
+      "piece": { "/": "bafk...commp" },
+    }
   },
   "fx": {
-    "join": { "/": "bafy...dequeue" }
+    "join": { "/": "bafy...pieceAccept" }
   },
   "meta": {},
-  "iss": "did:key:agg...",
+  "iss": "did:web:aggregator.web3.storage",
   "prf": []
 }
 ```
 
-See [`aggregate/add`](#aggregateadd) section to see the subsequent task.
+See the [`piece/accept`] section for the subsequent task.
 
-#### `aggregate/add`
+#### `piece/accept`
 
-When a piece request to be added is dequeued, an Aggregator MUST invoke `aggregate/add` to include it in an aggregate. The `storefront` requester identifier should be included in the invocation. See [schema](#aggregateadd-schema).
+An [Aggregator] MUST issue a receipt for the [`piece/accept`] task for the offered piece that was included in an aggregate. The receipt MUST contain an [`InclusionProof`] in the result and `fx.join` [effect] linking to [`aggregate/offer`] task, or an error detailing the reason.
 
-> `did:key:agg...` invokes capability from `did:key:agg...`
+> It is RECOMMENDED to never fail [`piece/accept`] as piece inclusion is a deterministic computation occurring on validated data.
 
 ```json
 {
-  "iss": "did:key:agg...",
-  "aud": "did:key:agg...",
-  "att": [{
-    "with": "did:key:agg...",
-    "can": "aggregate/add",
-    "nb": {
-      "piece": { "/": "commitment...car" }, /* commitment proof for piece */
-      "storefront": "did:web:web3.storage", /* storefront responsible for invocation (with of aggregate/queue) */
-      "group": "did:web:free.web3.storage", /* grouping of joining segments into an aggregate */
-    }
-  }],
-  "prf": [],
-  "sig": "..."
-}
-```
-
-An Aggregator MUST issue a signed receipt with the result of the task. An arranged aggregate for piece receipt looks like:
-
-```json
-{
-  "ran": "bafy...arrange",
+  "ran": "bafy...pieceAccept",
   "out": {
     "ok": {
-        "piece": { "/": "commitment...car" }, /* commitment proof for piece */
-        "aggregate": { "/": "commitment...aggregate-proof" }, /* commitment proof */
-        "path": "path-between-root-aggregate-and-piece"
+      /* commitment proof for piece */
+      "piece": { "/": "commitment...car" },
+      /* commitment proof for aggregate */
+      "aggregate": { "/": "commitment...aggregate" },
+      /** inclusion proof */
+      "inclusion": {
+        "tree": {
+          "path": [/** ... */],
+          "at": 4
+        },
+        "index": {
+          "path": [/** ... */],
+          "at": 7
+        }
+      }
     }
   },
```

Comment:
looks like we are missing here the effect to `aggregate/offer`?

Reply:
Good catch!

```suggestion
  "fx": {
    "join": { "/": "bafy...aggregateOffer" }
  },
```